third_party/cached_httpfs/http_file_cache.cpp

```diff
@@ -90,14 +81,31 @@ void CachedFileHandle::Read(void *buffer, idx_t length, idx_t offset) {
 	file->handle->Read((void *)buffer, length, offset);
 }
 
+LocalCacheFileSystem& HTTPFileCache::GetFS(const string &cache_dir) {
+	if (cached_fs) {
+		cached_fs->AssertSameCacheDir(cache_dir);
+		return *cached_fs;
+	}
+
+	{
+		lock_guard<mutex> lock(cached_fs_mutex);
```

Comment:
This is probably a bit overkill since `GetFS` is only called under a lock already, but feels more future-proof